mreza.py

```diff
@@ -0,0 +1,82 @@
+from Put import Put
+from Raskrsnica import Raskrsnica, Garaza
+from Vozilo import Vozilo
+from prepreka import Prepreka
+import random
+
+class Mreza: 
+    def __init__(self):
+        self.garaze = []
+        self.raskrsnice = []
+        self.putevi = []
+    
+    def dodajRaskrsnicu(self, id, koordinate, ulazni_putevi, izlazni_putevi):
+        raskrsnica = Raskrsnica(id, koordinate)
+        raskrsnica._ulazni_putevi = ulazni_putevi
+        raskrsnica._izlazni_putevi = izlazni_putevi
+        self.raskrsnice.append(raskrsnica)
+
+    def dodajGarazu(self, id, vreme_stvaranja, koordinate, izlazni_putevi):
+        self.dodajRaskrsnicu(id, koordinate, [id for _ in range(len(izlazni_putevi))], izlazni_putevi)
+        self.garaze.append(Garaza(id, koordinate, vreme_stvaranja, id))
+    
+    def dodajPut(self, id, duzina, max_brzina, id_raskrsnica, imaPrepreku=False):
+        if imaPrepreku:
+            prepreka = Prepreka()
+        else:
+            prepreka = None
+
+        self.putevi.append(Put(duzina, max_brzina, id, id_raskrsnica, prepreka))
+
+    def start(self, vehicles):
+        # Doesn't support threading for now
+        for vehicle in vehicles:
+            queue = []
+            queue.append(mreza.raskrsnice[0])
+            while len(queue) != 0:
+                front = queue[len(queue) - 1]
+                #print(f"{front._id}")
+                queue.pop()
+                nextRoad = random.choice(front._izlazni_putevi)
+                endConns = (front._id, nextRoad)
+                for put in mreza.putevi:
+                    if put._id_raskrsnica == endConns:
+                        putID, voziloID = put._id, vehicle.Uzmi_id()
+                        yield (putID, voziloID)
+
+                for raskrsnica in mreza.raskrsnice:
+                    if raskrsnica._id == nextRoad:
+                        queue.append(raskrsnica)
```

Comment:
zasto je ovo trostruka petlja? sta se ovde desava? Napraviti funkciju koja za jedno vozilo odredjuje gde ce ono ici pa to ubaciti u ovu fju. Start simulacije ne treba da bude u klasi mreza nego u napravite novu klasu simulacije koja vodi racuna o eventovima.
Razbiti ovo